packages/scene-composer/src/components/panels/SceneHierarchyPanel/components/SubModelTree/index.tsx

```diff
@@ -76,12 +85,14 @@ const SubModelTree: FC<SubModelTreeProps> = ({
         snapToFloor: true,
       },
       childRefs: [],
-      properties: [],
+      properties: {
+        subModelId: object3D.name,
+      },
     } as ISceneNodeInternal;
 
     appendSceneNodeInternal(node);
     setSceneNodeObject3DMapping(nodeRef, object3D); // Cache Reference
-  }, [object3D]);
+  }, [object3D, document.nodeMap]);
```

Comment:
Typically I'd flag this, but approving as this is made available via `useSceneDocument`.